### ISSUE TITLE
[BUG] Nix Flake wayland output is still broken

### ISSUE THREAD

**Issue by w-lfchen** [elkowar/eww#1036]
### Checklist before submitting an issue

- [X] I have searched through the existing [closed and open issues](https://github.com/elkowar/eww/issues?q=is%3Aissue) for eww and made sure this is not a duplicate
- [X] I have specifically verified that this bug is not a common [user error](https://github.com/elkowar/eww/issues?q=is%3Aissue+label%3Ano-actual-bug+is%3Aclosed)
- [X] I am providing as much relevant information as I am able to in this bug report (Minimal config to reproduce the issue for example, if applicable)

### Description of the bug

The `eww` and `eww-wayland` package outputs of the nix flake are currently broken on wayland.
This should have been fixed by #1032, but I overlooked something, I would like to apologize for my mistake there.
There are a few ways to resolve this, but I believe that I lack the experience to judge how this should be done.
1. Wait until NixOS/nixpkgs#289595 is merged and makes it into nixpkgs-unstable, then simply update the `flake.lock` file
2. This can be done right now: Revert [this line](https://github.com/elkowar/eww/blob/7bfd47eb8130f02f2a8f695c255df2f5302636b4/flake.nix#L47) to
`eww-wayland = final.eww.override { withWayland = true; };`, this will work temporarily until the aforementioned pr is merged and the lockfile is updated, then the `eww-wayland` output will no longer evaluate.
3. Change the options that cargo builds with. I'm sure that this is possible, but it can't simply be done by changing the overlay as far as I'm concerned and due to exams I currently don't have the time to look into this further. It would also be no longer needed once the pr is merged and the lockfile is updated.

### Reproducing the issue

Run `nix build` or `nix build .#eww-wayland`, nix will instruct cargo to build with the flags `--no-default-features --features=x11`, as specified [here](https://github.com/NixOS/nixpkgs/blob/b7aa428f8a74280ecc3d140f28f099a1e20b34eb/pkgs/applications/window-managers/eww/default.nix#L30-L33).

### Expected behaviour

Cargo should build with default features.

### Additional context

Once again, I am deeply sorry for overlooking this, I think it would be best to wait for the merge and then simply update the lockfile. My belief that this works is based on changing the `nixpkgs` url to `"github:coffee-is-power/nixpkgs/patch-1"`, which results in a correctly functioning binary.
Anything that resolves this issue should also result in #733, #775 and #817 being resolved.

On a different note, I believe that the current flake structure isn't that good due to the heavy reliance on `nixpkgs`, which is also why it's currently half broken. I admittedly don't know whether this is standard practice or not, I think it'd be a better idea to use the package in the overlay, not the other way around, but I can't rule out that I have made yet another mistake in understanding the inner works of all of this.

### Temporary fix

**Update: Use the pr as it should now be ready for a merge and is therefore in a usable state**
Use [this branch](https://github.com/w-lfchen/eww/tree/flake) from #1038 until the pr is merged:
```nix
{
  # url = "github:elkowar/eww";
  inputs.nixpkgs.follows = "nixpkgs";
  # use this until https://github.com/elkowar/eww/issues/1038 is merged
  url = "github:w-lfchen/eww/flake";
};
```
It shouldn't matter whether you override `nixpkgs` or not, please tell me if it does matter as this is not expected behavior.

**Comment by Cloudef** [comment]
Maybe you can consider using https://github.com/nix-community/naersk instead of rust-overlay

**Comment by w-lfchen** [comment]
[This line](https://github.com/Cloudef/nixos-flake/blob/1033d826ecd27f8f637d7c87edf1e0461c0e4589/flake.nix#L19C1-L19C44) is breaking it or will break in the future. It changes the `nixpkgs` input of the flake to be in line with the main one your flake depends on. If you remove it, my fork *should* work just fine.
This breaks my fork as all it does is change this nixpkgs input, which admittedly is a bit suboptimal as it results in you downloading nixpkgs at least twice but it is the easiest solution for the time being, I'll update the fix to point out that overriding the input is something you shouldn't do.
If you leave it in its current state, it could break once [pr 289595](https://nixpk.gs/pr-tracker.html?pr=289595) makes it into `nixos-unstable` and you update your lockfile due to the old flake then calling the eww-wayland output with an argument that no longer exists.

**Comment by w-lfchen** [comment]
@Cloudef You might want to check out my updated temporary fix, I'd appreciate it if you could comment under the pr whether it works for you (hopefully it does ^^')

**Comment by Cloudef** [comment]
>  If you remove it, my fork should work just fine.

Actually I did remove it but it did not work, so that's why I think it might be rust-overlay that eventually breaks it (and that also has nixpkgs follow)

**Comment by Cloudef** [comment]
Lemme try, as I just removed it and did not try with it with your recent changes.

**Comment by Cloudef** [comment]
It works even with rust-overlay in inputs and override now as well.

**Comment by w-lfchen** [comment]
Thank you for testing that, if I'm not mistaken it should work as long as the overlay is able to provide rust 1.76.0 since that is what eww needs
